### DISCUSSION TITLE
Adding and deleting events

### DISCUSSION THREAD

**Discussion by misulea** [vkurko/calendar#203] [Q&A] [answered]
Hello,

firstly, I would like to thank you for the great calendar. I wanted to ask if someone could send me an example code for adding and deleting an event, ideally with preparation for database. I came across a paragraph in the readme and also saw few examples here from discussion, but unfortunately, I still don't understand how to use the functionality.

Thanks in advance for any advice.

**Comment by vkurko** [comment]
Hello,

I would suggest loading data from the server using the [eventSources](https://github.com/vkurko/calendar#eventsources) option. In this option you can either specify a URL for loading events, or specify your own function that will provide the data. I would like to point out that in this way the data is not loaded in its entirety, but only those events that should be displayed in the current view, i.e. the calendar provides parameters for the start and end of the displayed period.

Then when you need to add or delete an event, you send a request to the server with the necessary parameters. And the next step is to call [refetchEvents](https://github.com/vkurko/calendar#refetchevents). This way the server will perform the necessary actions in the database, and after that the updated list of events will be re-loaded into the calendar.

**Reply by misulea** [reply]
Thank you for your answer. I saw that there is some preparation for the "add event" (when I drag somewhere on the empty space in the calendar, there is new event creating) but as soon as I release my mousebutton, the event immediately disappears. I would love to use this functionality - my idea of how it works is that after I release my MB, a pop-up or something will appear and I would be able to fill in the title etc.  With the deleting event, I would love to have a cross on the right top of the event and after clicking on it, the event will disappear both from visual and source.

**Reply by vkurko** [reply, marked as answer]
> I would love to use this functionality

Once you have _selected_ free time in the calendar, the [select](https://github.com/vkurko/calendar#select) callback is called, inside which you can create an event and add it to the calendar (as described above). Displaying a popup, interacting with your server to create an event or delete an event are all outside the scope of the Event Calendar library.

> I would love to have a cross on the right top of the event

As for displaying a cross inside an event, you can control the content that is displayed inside each event using the [eventContent](https://github.com/vkurko/calendar#eventcontent) option.

**Comment by ksiegel1923** [comment]
Hi! I am having the same problem as described above. I have figured out how to use the select callback but am then unsure of how to actually create an event within this callback. Does anyone have an example of how to do this?

**Reply by vkurko** [reply]
I have prepared a simple [demo](https://jsbin.com/tuqugeveve/edit?html,output) , where the selected period is immediately added to the list of events in the calendar.
